python/tests/test_highlevel.py

```diff
@@ -1441,6 +1441,33 @@ def verify_pairwise_diversity(self, ts):
     def test_pairwise_diversity(self, ts):
         self.verify_pairwise_diversity(ts)
 
+    def test_bad_node_order(self):
+        ts = tskit.TableCollection(1).tree_sequence()
+        for order in ["abc", 0, 1, False]:
+            with pytest.raises(ValueError, match="order"):
+                ts.nodes(order=order)
+
+    def test_node_order(self):
```

Comment:
This test looks generic, so could parametrize by the example tree sequences instead of just this one example?

Reply:
Yes, but I'm slightly worried that the example tree sequences all have nodes in time order (they were mostly generated by msprime), so I rolled my own version where I was sure the nodes were out-of-time-order.

Perhaps that should be added to the example tree sequences. I thought it wasn't worth testing the other examples though, as they would be likely to return nodes in the same order regardless of the order parameter.

Reply:
Right I missed that, sorry.

Perhaps you could check if the examples all have nodes in time order, and add this if not?

Reply:
The `decapitate` and `decapitate recomb` examples have a few nodes out of time order. I guess that's by "accident" rather than design. We could either add another deliberately out-of-order ts to the examples (but that'll increase the testing load) or perhaps add a "meta" test to check that the `get_example_tree_sequences()` function has some examples with out-of-time-order nodes?

Reply:
Just add in the example you've made. We need to review those examples anyway.

I don't like the idea of just testing this on a single example, we've hit errors before where corner cases get forgotten about. The examples contain some gnarly things like empty tree sequences which could affect, say, lexsort.

Reply:
Will do. Waiting on #2479 to be merged to make the test case easier.

Reply:
Example added in to the general set. Should be ready to merge.